app/src/main/java/com/android/sample/model/image/ImageRepositoryFirestore.kt

```diff
@@ -0,0 +1,145 @@
+                      .addOnSuccessListener {
+                        fileRef.downloadUrl.addOnSuccessListener { uri ->
+                          uploadedImageUrls.add(uri.toString())
+                          uploadCount++
+                          if (uploadCount == bitmaps.size) {
+                            firestore
+                                .collection("activities")
+                                .document(activityId)
+                                .update("images", uploadedImageUrls)
+                                .addOnSuccessListener { onSuccess(uploadedImageUrls) }
+                                .addOnFailureListener { onFailure(it) }
+                          }
+                        }
+                      }
+                      .addOnFailureListener { onFailure(it) }
+                }
+              }
+              .addOnFailureListener { onFailure(it) }
+        }
+        .addOnFailureListener { onFailure(it) }
+  }
+
+  override fun fetchProfileImageUrl(
+      userId: String,
+      onSuccess: (String) -> Unit,
+      onFailure: (Exception) -> Unit
+  ) {
+    val profilePicRef = storageRef.child("users/$userId/profile_picture.jpg")
+    profilePicRef.downloadUrl
+        .addOnSuccessListener { uri -> onSuccess(uri.toString()) }
+        .addOnFailureListener { onFailure(it) }
+  }
+
+  override fun fetchActivityImageUrls(
+      activityId: String,
+      onSuccess: (List<String>) -> Unit,
+      onFailure: (Exception) -> Unit
+  ) {
+    val firestore = FirebaseFirestore.getInstance()
+    val collection = firestore.collection("activities")
+
+    if (collection.document(activityId) == null) {
+      onFailure(IllegalStateException("Firestore collection reference is null"))
+      return
+    }
+
+    collection
+        .document(activityId)
+        .get()
+        .addOnSuccessListener { document ->
+          val imageUrls = document["images"] as? List<String> ?: emptyList()
+          onSuccess(imageUrls)
+        }
+        .addOnFailureListener { onFailure(it) }
+  }
+
+  override fun fetchActivityImagesAsBitmaps(
+      activityId: String,
+      onSuccess: (List<Bitmap>) -> Unit,
+      onFailure: (Exception) -> Unit
+  ) {
+    fetchActivityImageUrls(
+        activityId,
+        { urls ->
+          val bitmaps = mutableListOf<Bitmap>()
+          var successCount = 0
+
+          urls.forEach { url ->
+            val imageRef = FirebaseStorage.getInstance().getReferenceFromUrl(url)
+            imageRef
+                .getBytes(Long.MAX_VALUE)
+                .addOnSuccessListener { bytes ->
+                  val bitmap = BitmapFactory.decodeByteArray(bytes, 0, bytes.size)
+                  bitmaps.add(bitmap)
+                  successCount++
+                  if (successCount == urls.size) onSuccess(bitmaps)
+                }
+                .addOnFailureListener { onFailure(it) }
+          }
+        },
+        onFailure)
+  }
+}
```

Comment:
You can replace hardcoded strings with constants for better scalability and error prevention.
The fact that you have 25 and 100 is a bit confusing